src/reflect/scala/reflect/internal/Definitions.scala

```diff
@@ -1587,7 +1587,21 @@ trait Definitions extends api.StandardDefinitions {
       lazy val Boolean_not = definitions.Boolean_not
 
       lazy val Option_apply = getMemberMethod(OptionModule, nme.apply)
-      lazy val List_apply = DefinitionsClass.this.List_apply
+      private lazy val List_apply = DefinitionsClass.this.List_apply
```

Comment:
The field name `List_apply` seems quite misleading, should we rename it to `IterableFactory` (and also resolve as such)?